codegen/core/src/main/scala/com/lightbend/akkasls/codegen/ModelBuilder.scala

```diff
@@ -484,11 +488,32 @@ object ModelBuilder {
     if (ScalarType.isScalarType(resolvedName))
       ScalarTypeArgument(ScalarType(resolvedName))
     else {
-      val descriptor = lookupDescriptor(revolvedPackage, resolvedName, additionalDescriptors)
-      MessageTypeArgument(resolveFullyQualifiedMessageType(resolvedName, descriptor, additionalDescriptors))
+      lookupDomainDescriptor(revolvedPackage, resolvedName, additionalDescriptors)
+        .map { descriptor =>
+          MessageTypeArgument(resolveFullyQualifiedMessageType(resolvedName, descriptor, additionalDescriptors))
+        }
+        .getOrElse {
+          MessageTypeArgument(buildFullyQualifiedNameForPojos(revolvedPackage, resolvedName))
+        }
     }
   }
 
+  private def buildFullyQualifiedNameForPojos(packageName: String, className: String) = {
+    val fqn =
+      FullyQualifiedName(
+        protoName = "",
+        name = className,
+        PackageNaming(
+          protoFileName = "",
+          name = className,
+          protoPackage = "",
+          javaPackageOption = Some(packageName),
+          javaOuterClassnameOption = None,
+          javaMultipleFiles = true),
+        None)
+    fqn.copy(descriptorObject = Some(fqn))
+  }
```

Comment:
This is the hacky part. We need a `FullyQualifiedName` that is build from a simple package and class name. But our  `FullyQualifiedName` is very proto oriented, hence the `protoName = ""` etc.

Reply:
I'm working on another PR to fix this. I will send it on top of this one.